src/metabase/models/notification.clj

```diff
@@ -437,6 +437,14 @@
      (perms/current-user-has-application-permissions? :subscription))
     (current-user-can-read-payload? instance))))
 
+(defn can-unsubscribe?
+  "Check if the current user can unsubscribe from a notification."
+  [notification]
+  (or
+   (mi/superuser?)
+   (current-user-is-creator? notification)
```

Comment:
What does it mean to unsubscribe, if you're not currently a recipient?

Reply:
hmm, good question. I think you're right, we can simplify this to "current-user-is-recipient?"